plugins/flatpak/gs-flatpak.c

```diff
@@ -1825,20 +1801,23 @@ gs_plugin_refine_item_state (GsFlatpak *self,
 				return FALSE;
 			}
 		} else {
-			g_autoptr(GPtrArray) xrefs2 = NULL;
-			xrefs2 = flatpak_installation_list_installed_refs (installation,
-									   cancellable,
-									   error);
-			if (xrefs2 == NULL) {
+			g_autoptr(FlatpakInstalledRef) runtime_ref = NULL;
+			runtime_ref = get_installed_ref_for_app (self->installation, app,
+								 cancellable, &local_error);
+
+			if (runtime_ref != NULL) {
+				g_debug ("marking runtime %s as installed in the "
+					 "counterpart installation", gs_app_get_id (app));
+				gs_flatpak_set_metadata_installed (self, app, runtime_ref);
+				if (gs_app_get_state (app) == AS_APP_STATE_UNKNOWN)
+					gs_app_set_state (app, AS_APP_STATE_INSTALLED);
+			} else if (!g_error_matches (local_error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_INSTALLED)) {
+				/* if there was an error that wasn't simply that the app isn't installed, then
+				 * we need to return it*/
+				g_propagate_error (error, g_steal_pointer (&local_error));
 				gs_flatpak_error_convert (error);
 				return FALSE;
 			}
```

Comment:
`local_error` should be cleared here too

Reply:
Why? It's an auto_ptr and no code uses it anymore. I only cleaned it above because it was going to be reused.